.github/workflows/check_commit.yml

```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Check first line
         uses: gsactions/commit-message-checker@16fa2d5de096ae0d35626443bcd24f1e756cafee
         with:
-          pattern: '^((feat|fix|chore|refactor|style|test|docs|doc)\(\w+\)\:) .+$'
+          pattern: '^((feat|fix|chore|refactor|style|test|docs|doc)(\([\w\-_]+\))?\!?\:) .+$'
```

Comment:
Can you clarify the case you want to avoid ? If I undertand correctly you allow hypen and underscore within the parenthesis but not the exclamation mark right ?
What was the issue you've encountered ?

Reply:
it's something we did with Nico on the TFHE-rs repo to allow hypens, make the content of the parentheses optional and the ! is to allow indicating breaking change as conventional commits permits